src/data/utils.js

```diff
@@ -98,6 +103,33 @@ function zip(a, b) {
     return [x, b[i]];
   });
 }
+/**
+ * Do a pair-wise intersection of EpiDataRow by date.
```

Comment:
Maybe this is more straightforward code, but I am doing a similar thing for one pair of sensors, without depending on the ordering of the time_value, with the following function:

```
  let sensorListData = [
    { sensor: primary, data: primaryData },
    { sensor: secondary, data: secondaryData },
  ];

  $: sensorCorrelationData = mergeSensors();
  // for each time_value, merge data values across sensors.
  function mergeSensors() {
    const sensorDateMap = {};
    const sensorKeysMap = {}; // map from sensor key to sensor.
    sensorListData.map((sensorAndData) => {
      const { sensor, data } = sensorAndData;
      data.forEach((row) => {
        const time_value_key = String(row.time_value);
        if (!sensorDateMap[time_value_key]) {
          sensorDateMap[time_value_key] = { ...row };
        }
        const sensorKey = sensor.key;
        sensorKeysMap[sensorKey] = sensor;
        sensorDateMap[time_value_key][sensorKey] = row.value;
      });
    });
    return Object.values(sensorDateMap);
  }
```

I did it this way because I was merging rows for all the sensors, with slightly more complex code.

Reply:
If you think this operation is generalizable, I can export intersectEpiDataRow and have it return two new `EpiDataRow[]` using my previous `Map` based algorithm to be tolerant of order.

Perhaps additionally a new method that returns a `Map<Date, EpiDataRow>`.

Reply:
see also https://github.com/cmu-delphi/www-covidcast/blob/sgratzl/typescript/src/data/utils.ts (TS version for the types)

Reply:
I'll create another PR for additional utility methods.